src/services/publisher_service/mod.rs

```diff
@@ -177,6 +241,33 @@ async fn process_queued_messages(
     Ok(())
 }
 
+/// Process publishing with the threshold timeout
+#[instrument(skip(relay_http_client, metrics, analytics, notification))]
+async fn process_with_timeout(
+    execution_threshold: Duration,
+    notification: NotificationToProcess,
+    relay_http_client: Arc<Client>,
+    metrics: Option<&Metrics>,
+    analytics: &NotifyAnalytics,
+) -> crate::error::Result<()> {
+    match timeout(
+        execution_threshold,
+        process_notification(notification, relay_http_client.clone(), metrics, analytics),
+    )
+    .await
+    {
+        Ok(result) => {
+            if let Err(e) = result {
+                return Err(crate::error::Error::RelayPublishingError(e.to_string()));
```

Comment:
Why do we need to convert this to a String? Can we keep as the original error?

Reply:
I've fresh looked on this and seems adding an additional error type here is an overengineering. I've removed this type in favor of just an `error` type.